cmd/api/src/model/unified_graph.go

```diff
@@ -42,67 +42,85 @@ func NewUnifiedGraph() UnifiedGraph {
 
 // UnifiedNode represents a single node in a graph containing a minimal set of attributes for graph rendering
 type UnifiedNode struct {
-	Label      string    `json:"label"`
-	Kind       string    `json:"kind"`
-	ObjectId   string    `json:"objectId"`
-	IsTierZero bool      `json:"isTierZero"`
-	LastSeen   time.Time `json:"lastSeen"`
+	Label      string         `json:"label"`
+	Kind       string         `json:"kind"`
+	ObjectId   string         `json:"objectId"`
+	IsTierZero bool           `json:"isTierZero"`
+	LastSeen   time.Time      `json:"lastSeen"`
+	Properties map[string]any `json:"properties,omitempty"`
 }
 
 // UnifiedEdge represents a single path segment in a graph containing a minimal set of attributes for graph rendering
 type UnifiedEdge struct {
-	Source   string    `json:"source"`
-	Target   string    `json:"target"`
-	Label    string    `json:"label"`
-	Kind     string    `json:"kind"`
-	LastSeen time.Time `json:"lastSeen"`
+	Source     string         `json:"source"`
+	Target     string         `json:"target"`
+	Label      string         `json:"label"`
+	Kind       string         `json:"kind"`
+	LastSeen   time.Time      `json:"lastSeen"`
+	Properties map[string]any `json:"properties,omitempty"`
 }
 
-func FromDAWGSNode(node *graph.Node) UnifiedNode {
+func FromDAWGSNode(node *graph.Node, includeProperties bool) UnifiedNode {
 	var (
 		objectId   = getTypedPropertyOrDefault(node.Properties, common.ObjectID.String(), "")
 		label      = getTypedPropertyOrDefault(node.Properties, common.Name.String(), objectId)
 		systemTags = getTypedPropertyOrDefault(node.Properties, common.SystemTags.String(), "")
 		lastSeen   = getTypedPropertyOrDefault(node.Properties, common.LastSeen.String(), time.Now())
+		properties map[string]any
 	)
 
+	if includeProperties {
+		properties = node.Properties.Map
+	}
+
 	return UnifiedNode{
 		Label:      label,
 		Kind:       analysis.GetNodeKind(node).String(),
 		ObjectId:   objectId,
 		IsTierZero: strings.Contains(systemTags, ad.AdminTierZero),
 		LastSeen:   lastSeen,
+		Properties: properties,
 	}
 }
 
-func FromDAWGSRelationship(rel *graph.Relationship) UnifiedEdge {
-	return UnifiedEdge{
-		Source:   rel.StartID.String(),
-		Target:   rel.EndID.String(),
-		Kind:     rel.Kind.String(),
-		Label:    rel.Kind.String(),
-		LastSeen: getTypedPropertyOrDefault(rel.Properties, common.LastSeen.String(), time.Now()),
+// This is being used with slices.Map so it is necessary to return a closure
+func FromDAWGSRelationship(includeProperties bool) func(*graph.Relationship) UnifiedEdge {
+	return func(rel *graph.Relationship) UnifiedEdge {
```

Comment:
nice  😎

Reply:
Thankful that @ddlees helped me through aligning the types for use with the slices methods that are implemented with generics